### PR TITLE
auto-stop changes

### PR DESCRIPTION
**Pull request by mohitpavan**

### CODE REVIEW

File: lib/main.js [hunk 1]
```diff
@@ -75,13 +76,18 @@ function getTestAPI(validate) {
...
         if (testResult.message.statusCode == 401 || testResult.message.statusCode == 403) {
             var message = "Service Principal does not have sufficient permissions. Please assign "
                 + "the Load Test Contributor role to the service principal. Follow the steps listed at "
                 + "https://docs.microsoft.com/azure/load-testing/tutorial-cicd-github-actions#configure-the-github-actions-workflow-to-run-a-load-test ";
             throw new Error(message);
         }
+        if (testResult.message.statusCode != 200 && testResult.message.statusCode != 201) {
+            let testObj = yield util.getResultObj(testResult);
+            console.log(testObj ? testObj : util.ErrorCorrection(testResult));
+            throw new Error("Error in validating the jmx script.");
```
Comment (prativen) [inline lib/main.js:89]: Why error validating. Error fetching the test right?

File: lib/main.js [hunk 2]
```diff
@@ -111,7 +117,7 @@ function deleteFileAPI(filename) {
         var urlSuffix = "tests/" + testId + "/files/" + filename + "?api-version=2023-04-01-preview";
         urlSuffix = baseURL + urlSuffix;
         let header = yield map.getTestHeader();
-        let delFileResult = yield httpClient.del(urlSuffix, header);
+        let delFileResult = yield util.httpClientRetries(urlSuffix, header, 'del', 3, "");
         if (delFileResult.message.statusCode != 204) {
```
Comment (prativen) [inline lib/main.js:121]: is 204 the only success code here?

File: lib/util.js [hunk 3]
```diff
@@ -51,6 +53,37 @@ const validConditionList = {
...
+                httpResponse = yield httpClient.request(method, urlSuffix, content, header);
+            }
+            if (httpResponse.message.statusCode != undefined && [408, 404, 429, 502, 503, 504].includes(httpResponse.message.statusCode)) {
+                throw { message: httpResponse.message.statusMessage }; // throwing as message to catch it as err.message
+            }
+            return httpResponse;
+        }
+        catch (err) {
+            if (retries) {
+                let sleeptime = (5 - retries) * 1000 + Math.floor(Math.random() * 5001);
```
Comment (prativen) [inline lib/util.js:76]: how did we arrive at this function for sleep?

File: src/main.ts [hunk 4]
```diff
@@ -30,20 +30,26 @@ async function run() {
         await createTestAPI();
     }
     catch (err:any) {
+        console.log(err.message);
         core.setFailed(err.message);
     }
 }
 async function getTestAPI(validate:boolean) {
     var urlSuffix = "tests/"+testId+"?api-version=2023-04-01-preview";
```
Comment (prativen) [inline src/main.ts:38]: why are the files duplicated?

**Review by prativen — DISMISSED** [review]

File: lib/main.js (hunk 2 above)
Comment (mohitpavan) [reply lib/main.js:121]: yes for delete

File: src/main.ts (hunk 4 above)
Comment (mohitpavan) [reply src/main.ts:38]: this is not file, err.message only we can remove too, forgot to remove

File: lib/util.js (hunk 3 above)
Comment (mohitpavan) [reply lib/util.js:76]: 1st retries = 3, cuz i used recursion, when the number of retries remining getting decreased then we are increasing the await time

File: src/main.ts (hunk 4 above)
Comment (mohitpavan) [reply src/main.ts:37]: updated

**Review by radhikajain99 — APPROVED** [review]

**Pull request merged**